pkg/kf/commands/apps/integration_test.go

```diff
@@ -73,6 +74,13 @@ func TestIntegration_Push_update(t *testing.T) {
 		)
 		defer kf.Delete(ctx, appName)
 		checkEchoApp(ctx, t, kf, appName, 8087, ExpectedAddr(appName, ""))
+
+		kf.Push(ctx, appName,
+			"--path", filepath.Join(RootDir(ctx, t), "./samples/apps/helloworld"),
+		)
+		// XXX: it takes a moment after the app becomes ready to reconcile the
+		// routes
+		time.Sleep(30 * time.Second)
 		checkHelloWorldApp(ctx, t, kf, appName, 8088, ExpectedAddr(appName, ""))
```

Comment:
nit: Could we do this w/ retry logic instead of a sleep? It seems like other tests would benefit from this as well.

Reply:
This is a weird one. `checkHelloWorldApp` already does retries if the app isn't alive yet, but the app becomes ready before the routes are actually pointing at the new instance so the check fails because it sees the app as being alive but returning the wrong response. I opened #730 to track this.

Before this PR checkHelloWorldApp didn't actually check the value the app returned so the test passed even with the old echo app deployed.